rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/kustomize/KustomizeTemplateUtils.java

```diff
@@ -0,0 +1,153 @@
+    Artifact artifact = inputArtifacts.get(0);
+    String templatePath = getKustomizationPath(env,artifact);
+    try {
+      Path path = Paths.get(artifact.getReference());
+        artifact.setReference(path.toString());
+      for (String s : getFilesFromGithub(artifact)) {
+        Artifact a = new Artifact();
+        a.setReference(s);
+        a.setArtifactAccount(artifact.getArtifactAccount());
+        a.setCustomKind(artifact.isCustomKind());
+        a.setLocation(artifact.getLocation());
+        a.setMetadata(artifact.getMetadata());
+        a.setName(artifact.getName());
+        a.setProvenance(artifact.getProvenance());
+        a.setType(artifact.getType());
+        a.setUuid(artifact.getUuid());
+        a.setVersion(artifact.getVersion());
+        artifacts.add(a);
+      }
+    } catch (IOException e) {
+      log.error("Error setting references in artifacts from GitHub " + e.getMessage());
+    }
+    try {
+      for(Artifact ar : artifacts){
+        downloadArtifactToTmpFileStructure(env, ar);
+      }
+    } catch (IOException e) {
+      throw new IllegalStateException("Failed to fetch helm template: " + e.getMessage(), e);
+    }
+
+    List<String> command = new ArrayList<>();
+    command.add("kustomize");
+    command.add("build");
+    command.add( templatePath);
+
+
+      result.setCommand(command);
+
+    return result;
+  }
+
+  private HashSet<String> getFilesFromGithub(Artifact artifact) throws IOException {
+    HashSet<String> toEvaluate = new HashSet<>();
+    Path pathe = Paths.get(artifact.getReference());
+    Kustomization kustomization = kustomizationFileReader.getKustomization(artifact);
+    filesToDownload.add(kustomization.getReference());
+    if (kustomization.getResources() != null)
+      kustomization.getResources().forEach(f -> toEvaluate.add(f));
+    if (kustomization.getConfigMapGenerator() != null) {
+      kustomization
+              .getConfigMapGenerator()
+              .forEach(
+                      conf -> {
+                        conf.getFiles()
+                                .forEach(
+                                        f -> {
+                                          filesToDownload.add(pathe.resolve(f).toString());
+                                        });
+                      });
+    }
+    if (kustomization.getCdrs() != null)
+      kustomization.getCdrs().forEach(cdr -> toEvaluate.add(cdr));
+    if (kustomization.getGenerators() != null)
+      kustomization.getGenerators().forEach(gen -> toEvaluate.add(gen));
+    if (kustomization.getPatches() != null)
+      kustomization.getPatches().forEach(p -> toEvaluate.add(p.getPath()));
+    if (kustomization.getPatchesStrategicMerge() != null)
+      kustomization.getPatchesStrategicMerge().forEach(patch -> toEvaluate.add(patch));
+    if (kustomization.getPatchesJson6902() != null)
+      kustomization.getPatchesJson6902().forEach(json -> toEvaluate.add(json.getPath()));
```

Comment:
This is a lot of code to drop in here. I wonder if there's a way to encapsulate the logic for each of these into the classes representing them. Each class would be responsible for knowing which of it's properties can contain files and we could just call `kustomization.getDependencies()`. The Kustomization class would then call down into it's members, so on and so forth.

Reply:
moved to **Kustomization** class